Common/src/Common/Service/Table/Formatter/AbstractConversationMessage.php

```diff
@@ -33,6 +33,13 @@ public function format($row, $column = null): string
         // If createdBy (User) has a Team, they are an internal user.
         $internalCaseworkerTeam = (empty($row['createdBy']['team'])) ? '' : '<p class="govuk-caption-m">' . $senderName . '<br/>Caseworker Team</p>';
 
+        // If running on 'selfserve' remove caseworker's family name.
+        if (str_contains(__FILE__, 'selfserve') && $internalCaseworkerTeam) {
+            $caseworkerFamilyName = explode(' ', $senderName)[1];
+            $senderName = str_replace($caseworkerFamilyName, '', $senderName);
+            $internalCaseworkerTeam = str_replace($caseworkerFamilyName, '', $internalCaseworkerTeam);
+        }
+
```

Comment:
The `__FILE__` magic constant relies on a file path that contains `selfserve` which is not going always the case. For example, the file path I have for this file is: `/var/www/html/vendor/olcs/olcs-common/Common/src/Common/Service/Table/Formatter/AbstractConversationMessage.php`.

Instead, I recommend taking advantage of the inheritance that's going on in this area. The frontend applications will each have a concrete class that extends this _abstract_ class. Each one of those concrete classes could have its own `getSenderName` that formats appropriately. 

A few unit tests for this logic would also be ideal!

Give me a shout if you need a hand.